test/unit/plugins/rest_spec.js

```diff
@@ -222,9 +399,22 @@ describe("The Rest plugin", function () {
 		});
 
 		describe("with a valid type", function () {
+			var getAllInstancesStub;
 
 			before(function () {
-				createAnInstance(server);
+				getAllInstancesStub = Sinon.stub(instanceAdapter, "getAllInstances")
+				.returns(Bluebird.resolve([ {
+						id    : VALID_INSTANCE_ID,
```

Comment:
I don't think that this test is validating what it claims to be validating. Because the `getAllInstances()` method is stubbed and hard-coded to return a fixed instance list and the test case never validates that the correct type argument is sent to the method, we have no idea that the query is in fact processing correctly.